### PR TITLE
Fix exception on adding a new gamepad

### LINKED ISSUE #9591: Impossible to add new gamepads from inside the editor itself on Linux
### Expected behaviour

Being able to add new gamepads on the .gamepads form editor

### Actual behaviour

Not possible at all - it crashes as soon as you try adding a new one with the context menu `Add` option

### Steps to reproduce

- Open any gamepads file such as the builtin default.gamepads 
- Go into the gamepad list
- Right click to open the context menu
- Select Add
- Crash: clojure.lang.ExceptionInfo: Replace forbidden

<hr/>

<table>
<tr><td>Build time</td><td>2024-07-22T10:59:47.912057</td></tr>
<tr><td>Defold channel</td><td>editor-alpha</td></tr>
<tr><td>Defold editor sha</td><td>3be87d89fd5a93a63f527351fbedb84f8a875812</td></tr>
<tr><td>Defold engine sha</td><td>3be87d89fd5a93a63f527351fbedb84f8a875812</td></tr>
<tr><td>Defold version</td><td>1.9.1</td></tr>
<tr><td>Error</td><td><a href='https://sentry.io/organizations/defold/issues/?query=id%3A"3101c084391d49beb83ac57d654c3657"'>3101c084391d49beb83ac57d654c3657</a></td></tr>
<tr><td>GPU</td><td>NVIDIA GeForce RTX 2080 SUPER/PCIe/SSE2</td></tr>
<tr><td>GPU Driver</td><td>4.6.0 NVIDIA 560.35.03</td></tr>
<tr><td>Java version</td><td>17.0.5+8</td></tr>
<tr><td>OS arch</td><td>amd64</td></tr>
<tr><td>OS name</td><td>Linux</td></tr>
<tr><td>OS version</td><td>6.11.2-201.fsync.fc40.x86_64</td></tr>
<table>

### PR DESCRIPTION
**Pull request by vlaaad**
Fixes #9591

### CODE REVIEW

File: editor/project.clj [hunk 1]
```diff
@@ -263,6 +263,7 @@
                                                     [com.clojure-goes-fast/clj-async-profiler "0.5.1"]
                                                     [criterium "0.4.3"]
                                                     [lambdaisland/deep-diff2 "2.10.211"]
+                                                    [io.github.cljfx/dev "1.0.39"]
```
Comment (vlaaad) [inline editor/project.clj:266]: This lib improves cljfx stack-traces in dev:
```
clojure.lang.ExceptionInfo: Replace forbidden

Cljfx component stack:
  :text-formatter
  :text-field
  editor.cljfx-form-view/text-field
  clojure.lang.MultiFn@1f728db7
  :v-box
  :v-box
  :v-box
  clojure.lang.MultiFn@1f728db7
  :v-box
  :grid-pane
  :v-box
  editor.cljfx-form-view/section-view
  :v-box
  :scroll-pane
  cljfx.api/ext-on-instance-lifecycle
  cljfx.api/make-ext-with-props
  editor.cljfx-form-view/form-view

        at cljfx.mutator$fn__569.invokeStatic(mutator.clj:109)
        at cljfx.mutator$fn__569.invoke(mutator.clj:107)
        at cljfx.mutator$eval427$fn__428$G__416__441.invoke(mutator.clj:14)
        at cljfx.prop$replace_BANG_.invokeStatic(prop.clj:40)
        at cljfx.prop$replace_BANG_.invoke(prop.clj:38)
        ...
```

File: editor/src/clj/editor/cljfx_form_view.clj [hunk 2]
```diff
@@ -1526,52 +1532,57 @@
...
+                                          :header "The directory must reside within the project."})
+
+                                       :file-not-in-project
+                                       (dialogs/make-info-dialog
+                                         {:title "Invalid File"
+                                          :icon :icon/triangle-error
+                                          :header "The file must reside within the project."})))})
+                   (wrap-force-refresh view-id))}
+              (system/defold-dev?)
+              (assoc :fx.opt/type->lifecycle (requiring-resolve 'cljfx.dev/type->lifecycle)))
```
Comment (vlaaad) [inline editor/src/clj/editor/cljfx_form_view.clj:1585]: This diff is mostly indentation due to wrapping in `cond->`, here is the actual change: in dev, use `cljfx/dev` lib for better error reporting.

File: editor/src/clj/editor/cljfx_form_view.clj [hunk 3]
```diff
@@ -231,10 +232,15 @@
 
 ;; region string input
 
+(defn- text-formatter [props]
+  {:fx/type fx/ext-recreate-on-key-changed
+   :key (:value-converter props)
+   :desc (assoc props :fx/type fx.text-formatter/lifecycle)})
```
Comment (vlaaad) [inline editor/src/clj/editor/cljfx_form_view.clj:238]: The fix: re-create the whole `text-formatter` object when the `value-converter` changes.

**Review by matgis — APPROVED** [review]
Ship it! ❤️

**Pull request merged**